posthog/api/test/test_capture.py

```diff
@@ -504,22 +492,25 @@ def test_batch_token_not_set(self):
             ),
         )
 
-    def test_batch_distinct_id_not_set(self):
+    @patch("statshog.defaults.django.statsd.incr")
+    def test_batch_distinct_id_not_set(self, statsd_incr):
         response = self.client.post(
             "/batch/",
             data={"api_key": self.team.api_token, "batch": [{"type": "capture", "event": "user signed up",},],},
             content_type="application/json",
         )
 
-        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
-        self.assertEqual(
-            response.json(),
-            self.validation_error_response(
-                "You need to set user distinct ID field `distinct_id`.", code="required", attr="distinct_id"
-            ),
-        )
+        # An invalid distinct ID will not return an error code, instead we will capture an exception
```

Comment:
Now I see. Is this a change that might need coordination with SDKs, docs and so on?

Reply:
It shouldn't need coordination with SDKs, docs might be a good addition. 

While this sounds like a big change (our most important API behaves differently after this), it's not that bad when you consider that this was actually a bug.

If you sent us 999 valid events, and the last event was an invalid one, we'd send you back a 4xx response. That might lead a client to retry sending these (think platforms like RudderStack or Segment), leading to a lot of duplicates.

Reply:
Anyway, none of our SDKs rely on these particular errors. But indeed it should be made clear on the docs that if you send us a batch of events we will not tell you if some events in there are invalid.

Reply:
Here's a start on docs: https://github.com/PostHog/posthog.com/pull/2178